tide/core/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
 set(TIDECORE_LINK_LIBRARIES
   PUBLIC
-    Deflect
+    DeflectServer
```

Comment:
technically, deflect should be both, private and optional. Out of scope of this PR, but still :)

Reply:
Yeah, but there is no benefit in making Deflect private here, tide is a top-level application and the deflect types are used in multiple places... I agree in principle for the optional part, but not having deflect would drastically reduce Tide's feature set. Deflect's only requirement is libjpegturbo, which I think is not a big blocker. In fact even this dependency could be optional (users would get uncompressed streaming only), but I think we forced it required a long time ago...